webapp/src/js/services/send-message.js

```diff
@@ -64,7 +64,14 @@ angular.module('inboxServices').factory('SendMessage',
         include_docs: true,
         key: recipient.doc._id
       }).then(function(results) {
-        return results.rows;
+        return results.rows
+          .filter(function(row) {
+            return row.doc.contact;
+          })
+          .map(function(row) {
+            row.doc.descendant = true;
```

Comment:
Try not to modify db docs with properties that you don't want to end up in the database - it's all too easy for a future change to persist these docs and cause subtle bugs. We've had an issue before where `expanded: true` which is used for expanding/collapsing in the UI got persisted to the db...

I think this is made unnecessarily complicated by `mapRecipients` trying to handle various inputs. I suggest you modify this so that...

```js
    var mapRecipient = function(contact, phone) {
      if (phone) {
        return {
          contact: contact,
          phone: phone
        };
      }
    };

    var descendants = function(recipient) {
      return DB().query('medic-client/contacts_by_parent', {
        include_docs: true,
        key: recipient.doc._id
      }).then(function(results) {
        return results.rows.map(function(row) {
          var doc = row.doc;
          var phone = doc.contact && doc.contact.phone;
          return mapRecipient(doc, phone);
        });
      });
    };

    var hydrate = function(recipients) {
      var ids = recipients.map(function(recipient) {
        return recipient.doc._id;
      });
      return DB().allDocs({ include_docs: true, keys: ids }).then(function(results) {
        return results.rows.map(function(row) {
          var doc = row.doc;
          var phone = doc.phone || (doc.contact && doc.contact.phone);
          return mapRecipient(doc, phone);
        });
      });
    };

    var resolvePhoneNumbers = function(recipients) {
      //TODO: do we want to attempt to resolve phone numbers into existing contacts?
      // users will have already got that suggestion in the send-message UI if
      // it exists in the DB
      return recipients.map(function(recipient) {
        var phone = recipient.text || // from select2
                    recipient.doc.contact.phone; // from LHS message bar
        return mapRecipient(null, phone);
      });
    };
```

Now you don't need the `mapRecipients` (plural) function at all because the recipients are already in the right format.

What do you think?